### PR TITLE
Added handling for non xsoar packs

### PR DESCRIPTION
**Pull request by ShahafBenYakir**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/29574
fixes: https://github.com/demisto/etc/issues/28279

### CODE REVIEW

File: Tests/scripts/utils/content_packs_util.py [hunk 1]
```diff
@@ -50,5 +50,26 @@ def should_test_content_pack(pack_name: str) -> bool:
...
     """
+    if not pack_name:
+        return False
     pack_path = os.path.join(PACKS_DIR, pack_name)
-    return pack_name not in SKIPPED_PACKS and is_pack_certified(pack_path)
+    return pack_name not in SKIPPED_PACKS and is_pack_xsoar_supported(pack_path)
+
+
+def get_test_pack_name(test_id: str, id_set: json) -> str:
+    """Returns a the pack name containing the given test
```
Comment (ShahafBenYakir) [inline Tests/scripts/utils/content_packs_util.py:60]:
```suggestion
    """
    Returns the pack name containing the given test
```

File: Utils/trigger_content_nightly_build.sh [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
-_branch=$1
-_circle_token=$2
+_circle_token=$1
+[ -n "$2" ] && _branch="$2" || _branch="$(git branch | sed -n -e 's/^\* \(.*\)/\1/p')"
```
Comment (yaakovi) [inline Utils/trigger_content_nightly_build.sh:4]:
```suggestion
[ -n "$2" ] && _branch="$2" || _branch="$(git branch --show-current)"
```

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 3]
```diff
@@ -986,21 +989,50 @@ def remove_ignored_tests(tests: set, content_packs: set) -> set:
...
+                tests_that_should_not_be_tested.add(test)
+
+    if tests_that_should_not_be_tested:
+        logging.info('The following test playbooks are not supported and will not be tested: \n{} '.format(
+            '\n'.join(tests_that_should_not_be_tested)))
+        tests.difference_update(tests_that_should_not_be_tested)
+    return tests
+
+
+def filter_tests(tests: set, id_set: json) -> set:
```
Comment (Itay4) [inline Tests/scripts/collect_tests_and_content_packs.py:1010]:
```suggestion
def filter_tests(tests: set, id_set: dict) -> set:
```

Comment (Itay4) [inline Tests/scripts/collect_tests_and_content_packs.py:1010]: add docstring

Comment (ShahafBenYakir) [reply Tests/scripts/collect_tests_and_content_packs.py:1010]: Shtak

File: Utils/trigger_content_nightly_build.sh [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
-_branch=$1
-_circle_token=$2
+_circle_token=$1
```
Comment (Itay4) [inline Utils/trigger_content_nightly_build.sh:3]: why did you change the order of the args?

File: Tests/configure_and_test_integration_instances.py [hunk 5]
```diff
@@ -1311,7 +1313,7 @@ def main():
     else:
         installed_content_packs_successfully = True
 
-    tests_for_iteration = get_tests(build.server_numeric_version, build.tests)
+    tests_for_iteration = get_tests(build)
     new_integrations, modified_integrations = get_changed_integrations(build)
```
Comment (Itay4) [inline Tests/configure_and_test_integration_instances.py:1317]: please merge from master - @hod-alpert modified this code

File: Tests/configure_and_test_integration_instances.py [hunk 6]
```diff
@@ -958,22 +958,24 @@ def restart_server_legacy(server):
         logging.exception('Legacy SSH restart demisto failed')
 
 
-def get_tests(server_numeric_version: str, tests: list) -> list:
+def get_tests(build):
```
Comment (Itay4) [inline Tests/configure_and_test_integration_instances.py:961]: please add type hints

File: Tests/test_content.py [hunk 7]
```diff
@@ -644,10 +646,8 @@ def extract_filtered_tests():
     with open(FILTER_CONF, 'r') as filter_file:
         filtered_tests = filter_file.readlines()
         filtered_tests = [line.strip('\n') for line in filtered_tests]
```
Comment (Itay4) [inline Tests/test_content.py:648]: do it a one-liner:
```python
filtered_tests = [line.strip('\n') for line in filter_file.readlines()]
```

File: Tests/configure_and_test_integration_instances.py [hunk 8]
```diff
@@ -958,22 +958,24 @@ def restart_server_legacy(server):
...
+    server_numeric_version: str = build.server_numeric_version
+    tests: dict = build.tests
     if Build.run_environment == Running.CIRCLECI_RUN:
-        filtered_tests, filter_configured, run_all_tests = extract_filtered_tests()
-        if run_all_tests:
-            logging.warning(f'Not running instance tests when {RUN_ALL_TESTS_FORMAT} is turned on')
+        filtered_tests = extract_filtered_tests()
+        if build.is_nightly:
+            # skip test button testing
+            logging.warning('Not running instance tests in nightly flow')
```
Comment (Itay4) [inline Tests/configure_and_test_integration_instances.py:976]: lets change logging level, this shouldnt be a warning

File: Tests/private_build/run_content_tests_private.py [hunk 9]
```diff
@@ -22,7 +22,7 @@
     print_test_summary, update_test_msg, turn_off_telemetry, \
     create_result_files, get_all_tests, get_instances_ips_and_names, get_server_numeric_version, \
     initialize_queue_and_executed_tests_set, get_test_records_of_given_test_names, \
-    extract_filtered_tests, load_conf_files, set_integration_params, collect_integrations, notify_failed_test,\
+    extract_filtered_tests, load_conf_files, set_integration_params, collect_integrations, notify_failed_test, \
```
Comment (Itay4) [inline Tests/private_build/run_content_tests_private.py:25]: how did we test the changes in this file? are @amshamah419 and @guykeller aware?

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 10]
```diff
@@ -78,6 +78,7 @@ def get_tested_integrations_for_collected_tests(self, collected_tests):
 
         return tested_integrations
 
+    # This function is the same function exactly as 'get_content_pack_name_of_test' and therefore should be removed
```
Comment (Itay4) [inline Tests/scripts/collect_tests_and_content_packs.py:81]: lets open an issue for this and add TODO here

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 11]
```diff
@@ -357,7 +358,7 @@ def find_tests_and_content_packs_for_modified_files(modified_files, conf=deepcop
 
     if len(missing_ids) > 0:
         test_string = '\n'.join(missing_ids)
-        message = "You've failed to provide tests for:\n{0}".format(test_string)
+        message = "Was not able to find tests for:\n{0}".format(test_string)
```
Comment (Itay4) [inline Tests/scripts/collect_tests_and_content_packs.py:361]: you are too nice

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 12]
```diff
@@ -963,18 +962,18 @@ def get_modified_packs(files_string):
...
+def remove_ignored_tests(tests: set, id_set: dict) -> set:
     """Removes test playbooks, which are in .pack-ignore, from the given tests set
 
     Args:
         tests (set): Tests set to remove the tests to ignore from
-        content_packs (set): Content packs from which to check if test should be ignored
+        id_set (dict): The id set object
 
-    Return:
+    Return:remove_ignored_tests
```
Comment (Itay4) [inline Tests/scripts/collect_tests_and_content_packs.py:972]: ?

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 13]
```diff
@@ -986,21 +985,56 @@ def remove_ignored_tests(tests: set, content_packs: set) -> set:
...
+    if tests_that_should_not_be_tested:
+        logging.debug('The following test playbooks are not supported and will not be tested: \n{} '.format(
+            '\n'.join(tests_that_should_not_be_tested)))
+        tests.difference_update(tests_that_should_not_be_tested)
+    return tests
+
+
+def filter_tests(tests: set, id_set: json) -> set:
+    """
+    Filter tests out from the test set if they are a.Ignored b.Non XSOAR and non-supported packs;
```
Comment (Itay4) [inline Tests/scripts/collect_tests_and_content_packs.py:1008]: lets stick to google style docstring

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 14]
```diff
@@ -1023,11 +1057,33 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
...
-    packs_to_install = packs_to_install.union(get_content_pack_name_of_test(tests, id_set))
-
     if is_conf_json:
         tests = tests.union(get_test_from_conf(branch_name, conf))
 
+    if changed_common:
+        tests.add('TestCommonPython')
+
+    # get all modified packs - not just tests related
+    # EDIT: this fix is bad and need to be changed in the source (get_modified_files_for_testing) and not after all
```
Comment (Itay4) [inline Tests/scripts/collect_tests_and_content_packs.py:1067]: lets open issue and add TODO here

File: Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py [hunk 15]
```diff
@@ -440,6 +458,9 @@ def test_changed_runnable_test__unmocked_get_modified_files(self, mocker):
                 with_scripts=fake_script['id_set'],
                 with_test_playbook=fake_test_playbook['id_set']
             )
+            # Creating pack metadata
```
Comment (Itay4) [inline Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py:461]: remove comments

File: Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py [hunk 16]
```diff
@@ -575,7 +596,7 @@ def create_get_modified_files_ret(modified_files_list=None, modified_tests_list=
 
 
 def get_mock_test_list(minimum_server_version=TWO_BEFORE_GA_VERSION, get_modified_files_ret=None, mocker=None,
-                       git_diff_ret=''):
+                       git_diff_ret='', ):
```
Comment (Itay4) [inline Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py:599]: revert this change

File: Tests/scripts/utils/tests/content_packs_util_test.py [hunk 17]
```diff
@@ -65,4 +68,14 @@ def test_should_test_content_pack(pack_name, expected):
         - Case A: Verify pack should be tested
         - Case B: Verify pack should not be tested
     """
+    # Creating temp dirs
+    packs_dir = tmp_path / PACKS_DIR
+    packs_dir.mkdir()
+    pack = packs_dir / pack_name
+    pack.mkdir()
```
Comment (Itay4) [inline Tests/scripts/utils/tests/content_packs_util_test.py:75]: no need to create the directories in 2 seperate calls, you can do the following:
```python
pack = tmp_path / PACKS_DIR / pack_name
pack.mkdir(parents=True)
```

**Comment by Itay4** [comment]
@ShahafBenYakir how did we test this? did we trigger nightly build against these changes?

File: Tests/configure_and_test_integration_instances.py (hunk 5 above)
Comment (ShahafBenYakir) [reply Tests/configure_and_test_integration_instances.py:1317]: Old news.

File: Tests/private_build/run_content_tests_private.py (hunk 9 above)
Comment (ShahafBenYakir) [reply Tests/private_build/run_content_tests_private.py:25]: The change here is a space added automatically

File: Utils/trigger_content_nightly_build.sh (hunk 4 above)
Comment (ShahafBenYakir) [reply Utils/trigger_content_nightly_build.sh:3]: I need to check if the `_branch` argument was given or not. Since we pass the value anonymously, if the order was as before and the user did not give a branch `_branch` would have the API KEY value.
Another solution was to add a check for the count of arguments `$#`.

File: Tests/scripts/utils/tests/content_packs_util_test.py (hunk 17 above)
Comment (ShahafBenYakir) [reply Tests/scripts/utils/tests/content_packs_util_test.py:75]: Where did you find this? Didn't manage to find anything like this in docs.

File: Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py (hunk 16 above)
Comment (ShahafBenYakir) [reply Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py:599]: Woops

File: Tests/scripts/collect_tests_and_content_packs.py (hunk 10 above)
Comment (ShahafBenYakir) [reply Tests/scripts/collect_tests_and_content_packs.py:81]: https://github.com/demisto/etc/issues/30672

**Review by Itay4 — APPROVED** [review]

File: Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py [hunk 18]
```diff
@@ -588,6 +606,16 @@ def get_mock_test_list(minimum_server_version=TWO_BEFORE_GA_VERSION, get_modifie
     return tests, content_packs
 
 
+def create_temp_dir_with_metadata(path, pack_name, pack_metadata_content):
+    packs_dir = path / PACKS_DIR
+    packs_dir.mkdir()
+    pack = packs_dir / pack_name
```
Comment (Itay4) [inline Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py:612]: FIX

Comment (ShahafBenYakir) [reply Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py]: fixed
